### PR TITLE
fix: fix nested hparams with grid [DET-6045]

### PR DESCRIPTION
**Pull request by rb-determined-ai**
## Description

Nested hparams weren't working properly with grid search.

## Test Plan

Added a test which failed without this fix.

### CODE REVIEW

**Comment by rb-determined-ai** [comment]
@liamcli as I said before, I rewrote the grid sampling logic almost entirely.  I think it is simpler as-written, but I can go back and fix _just_ the bug if you'd like.

File: master/pkg/searcher/grid_test.go [hunk 1]
```diff
@@ -44,17 +46,34 @@ func TestGridFunctionality(t *testing.T) {
 
 func TestHyperparameterGridMethod(t *testing.T) {
 	dParam := expconf.DoubleHyperparameter{RawMaxval: 2.0, RawCount: ptrs.IntPtr(5)}
-	assert.Equal(t, len(grid(expconf.Hyperparameter{RawDoubleHyperparameter: &dParam})), 5)
+	assert.Equal(t,
+		len(getGridAxes([]string{"x"}, expconf.Hyperparameter{RawDoubleHyperparameter: &dParam})),
```
Comment (liamcli) [inline master/pkg/searcher/grid_test.go:50]: need to index 0 on the result of getGridAxes for this assert as well as others in this test.

Comment (rb-determined-ai) [reply master/pkg/searcher/grid_test.go:50]: makes sense, thank you

**Review by liamcli — APPROVED** [review]
lgtm, thanks for fixing it!

**Pull request merged**